### ISSUE TITLE
Support a 1 service : N port abstraction

### ISSUE THREAD

**Issue by armon** [hashicorp/consul#287]
Many services expose more than one port. This currently is done by exporting multiple service entries, but can be done more nicely.

**Comment by wwalker** [comment]
+1  Have multiple multi-port apps here

**Comment by niclashoyer** [comment]
@armon is there any roadmap for this?

**Comment by ryanuber** [comment]
@niclashoyer we will likely get this added once the 0.5.0 release is complete. We are trying to lock down on features to get that release out the door since we already slipped the intended release date.

**Comment by sigil66** [comment]
+1

**Comment by slackpad** [comment]
Closing this as a duplicate of https://github.com/hashicorp/consul/issues/1202, since there's some slightly more recent discussion going on there.
